torchtune/training/_compile.py

```diff
@@ -0,0 +1,70 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+import os
+from typing import Union
+
+from torchtune.modules import (
+    CEWithChunkedOutputLoss,
+    TiedEmbeddingTransformerDecoder,
+    TransformerDecoder,
+)
+from torchtune.utils import get_logger, torch_version_ge
+
+log = get_logger("INFO")
+
+
+def compile_model(
+    model: Union[TransformerDecoder, TiedEmbeddingTransformerDecoder],
+    verbose: bool = True,
+) -> None:
+    """
+    Utility to compile a transformer model inplace. On PyTorch nightlies we use per-layer compile
+    to reduce compile times. Otherwise we compile the full model, which takes longer.
+
+    Args:
+        model (Union[TransformerDecoder, TiedEmbeddingTransformerDecoder]): A transformer model to compile.
+        verbose (bool): Whether to log compile info. Default: True
+    Returns:
+        None
+    """
+    backend = os.environ.get("TORCH_COMPILE_BACKEND", "inductor")
+    if torch_version_ge("2.5.0"):
```

Comment:
: (

Reply:
Come 2.5 our codebase will be so beautiful

Reply:
Oh but then we will just have a bunch of 2.6 nightly features..